docs/hd.md

```diff
@@ -0,0 +1,124 @@
+# Cardano SL HD Wallets
+
+## General scheme
+    
+HD wallet has tree-based structure.
+The root of the tree is user's public-secret key pair.
+Each node in the tree corresponds to public-secret key pair.
+
+Each child node is identified by an index of this child,
+so child's public and secret keys can be _derived_ from parent
+by this index using parent's public-secret key pair.
```

Comment:
I think the main problem with this documentation is that it's not crystal clear about the relationship between a private key, a public key, an extended public key and an extended private key is.

To begin with, I think we should try to pick a naming scheme as close as possible as [BIP-32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#Specification_Key_derivation), so people which are familiar with "standard" HD wallets won't be too confused.

I'm even OK with the fact we pick different names, but we should be super precise about:

* When do we mean a `public key` intended as a point on an elliptic curve?
* When do we mean an `extended public key` intended as the tuple (public key, chain code) ?
* When do we mean a `private key` intended as random bits that, when multiplied by the fixed-point G on the elliptic curve, yields a `public key`?
* When do we mean an `extended private key` intended as the tuple (private key, chain code)?

Unfortunately the confusion stems from the fact that, in the code, we have a `newtype` wrapper around an `XPub` which is called `PublicKey` (if memory does not betray me).